rate_counter.go

```diff
@@ -237,22 +240,22 @@ func (this *RateCounterSnapshot) Snapshot() RateCounter {
 	return this
 }
 
-type NilRateCounter struct {}
+type NilRateCounter struct{}
 
-func (this NilRateCounter)	Mark(int64) {
+func (this NilRateCounter) Mark(int64) {
```

Comment:
nit: None of these receivers need to be named (and `this` should never be used as a receiver name).